src/test/java/de/tum/in/www1/artemis/exercise/programmingexercise/ProgrammingExerciseResultBambooIntegrationTest.java

```diff
@@ -77,6 +77,20 @@ void shouldUpdateTestCasesAndResultScoreFromSolutionParticipationResultWithFaile
         programmingExerciseResultTestService.shouldUpdateTestCasesAndResultScoreFromSolutionParticipationResult(notification, true);
     }
 
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "student1", roles = "USER")
+    void shouldHandleTestCasesWithLongNames() throws JsonProcessingException {
+        String name1 = "a".repeat(254);
+        String name2 = "b".repeat(255);
+        String name3 = "c".repeat(256);
+        var notification = ProgrammingExerciseFactory.generateBambooBuildResult(Constants.ASSIGNMENT_REPO_NAME, null, null, null, List.of(name1, name2), List.of(name3),
+                new ArrayList<>());
```

Comment:
Can this be moved into `programmingExerciseResultTestService.shouldHandleTestCasesWithLongNames` to avoid the duplication with the Jenkins test?
(Assuming this code should not do anything that affects the order of the mocking.)

Reply:
We need the tests for `generateBambooBuildResult` and the corresponding method for Jenkins. So I don't really see how we can easily avoid this duplication without actually writing more lines of code

Reply:
Ah, missed the specific part for the build result generation. Never mind then.
:+1: